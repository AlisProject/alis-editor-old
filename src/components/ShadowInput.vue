<template>
  <textarea
    @keydown="handleKeydown"
    @paste="handlePaste"
  ></textarea>
</template>

<script lang="ts">
import Vue from 'vue'
import { createDataURIImage } from '../utils/createImage'

export default Vue.extend({
  methods: {
    handleKeydown(event: KeyboardEvent) {
      if (event.keyCode !== 8) {
        return
      }
      this.$emit('delete', event)
    },
    handlePaste(event: any) {
      ;(async () => {
        const src = await createDataURIImage(event)
        this.$emit('addimageuri', src)
      })()
    }
  }
})
</script>

<style scoped>
textarea {
  font-size: 20px !important;
  color: transparent !important;
  background: transparent !important;
  resize: none !important;
  opacity: 0 !important;
  outline: none !important;
  display: block !important;
  position: absolute !important;
  z-index: 100000 !important;
  width: 100% !important;
  height: 100% !important;
  left: 0 !important;
  top: 0 !important;
  border: 0 !important;
  padding: 0 !important;
  margin: 0 !important;
}
</style>
