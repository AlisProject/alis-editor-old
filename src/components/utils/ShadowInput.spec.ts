import { mount } from '@vue/test-utils'
import ShadowInput from './ShadowInput.vue'

describe('ShadowInput.vue', () => {
  test('renders correctly', () => {
    const wrapper = mount(ShadowInput)
    expect(wrapper.element).toMatchSnapshot()
  })
})
