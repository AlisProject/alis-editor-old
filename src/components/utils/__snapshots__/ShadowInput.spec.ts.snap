// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShadowInput.vue renders correctly 1`] = `
<textarea
  class="shadow-input"
/>
`;
